src/NUnitFramework/tests/Internal/NamespaceTests.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using NUnit.Framework.Api;
+
+namespace NUnit.Framework.Internal
+{
+    [TestFixture]
+    public class NamespaceTests
+    {
+        [Test]
+        public void AllExportedNameSpacesAreNotSystem()
+        {
+#if !NETCOREAPP1_1
+            var exportedTypes = Assembly.GetAssembly(typeof(FrameworkController)).GetExportedTypes();
+#else
+            var exportedTypes = typeof(FrameworkController).GetTypeInfo().Assembly.GetExportedTypes();
+#endif
```

Comment:
I'm not sure - but does the second version of this not work on all platforms? 🙂

Reply:
Just looked deeper, looks like we have an extension method implementation of GetTypeInfo() in ReflectionExtensions.cs to help with this case? I thought the idea of that is that it would allow us to use the same API across all platforms - I’m not 100% sure if it covers this case or not though, sorry! 😊

Reply:
If I'm wrong, lets get this merged. 😄

Reply:
I've been starting to think we should honestly put our TypeExtensions in System.Reflection so that we don't need both namespaces imported. Or maybe namespaceless. It's so finicky.

Reply:
Great - thanks for that! I wasn't sure myself! 😅